04-onion-routing.md

```diff
@@ -269,7 +272,81 @@ The writer:
 The reader:
   - MUST return an error if `amt_to_forward` or `outgoing_cltv_value` are not present.
 
-The requirements for the contents of these fields are specified [above](#legacy-hop_data-payload-format).
+The requirements for the contents of these fields are specified [above](#legacy-hop_data-payload-format)
+and [below](#basic-multi-part-payments).
+
+### Basic Multi-Part Payments
+
+If the final node receives an onion packet with the `basic_mpp` field,
+then the payment MAY be a "base" atomic multipath payment.  Such
+"base" atomic multipath payments will use the same `payment_hash` for
+all paths.
+
+The `amt_to_forward` value will be the amount for this partial payment
+only.  The `basic_mpp` flag is a promise by the ultimate sender that
+the rest of the payment will follow in succeeding HTLCs; we call these
+HTLCs, which have the same preimage, an "HTLC set".
+
+#### Requirements
+
+The writer:
+  - MUST NOT include `basic_mpp` for any non-final node.
+  - if `basic_mpp` feature was not set in the invoice:
+    - MUST NOT include `basic_mpp` for the final node.
+  - otherwise:
+    - MAY include `basic_mpp` for the final node.
+    - if it does include `basic_mpp`:
+      - MAY send more than one HTLC to pay the invoice.
+      - MUST use the same `payment_preimage` on all HTLCs in the set.
+      - MUST set `total_msat` to the amount it wishes to pay.
+      - SHOULD send all payments at approximately the same time.
+      - SHOULD try to use diverse paths to the recipient for each HTLC.
+      - SHOULD retry and/or re-divide HTLCs which fail.
+      - if the invoice specifies an `amount`:
+          - MUST set `total_msat` to at least that `amount`, and less
+            than or equal to twice `amount`.
+      - MUST ensure that the total `amount_msat` of the HTLC set which arrive at the payee
+        is greater or equal to `total_msat`.
+      - MUST ensure that no (incomplete) non-failed subset of the HTLC set adds
+        to a total `amount_msat` greater or equal to `total_msat`.
```

Comment:
What does this mean exactly? If the sender has enough htlcs in flight to pay the invoice but the htlcs are not settled, they shouldn't add another htlc?

Reply:
Exactly that's it. If you've sent an HTLC set that pays the total amount, you should wait for (some of) them to fail or settle. If you send another HTLC you will end up paying more than what you intended to pay.

Reply:
No strict subset.  This has been a source of confusion, though, so we need to reword it.